lib/clif-backend/src/resolver.rs

```diff
@@ -96,59 +96,169 @@ impl FuncResolverBuilder {
 
     pub fn new(
         isa: &dyn isa::TargetIsa,
-        function_bodies: Map<LocalFuncIndex, ir::Function>,
+        function_bodies: Map<LocalFuncIndex, (ir::Function, (u32, u32))>,
         info: &ModuleInfo,
-    ) -> CompileResult<(Self, HandlerData)> {
+    ) -> CompileResult<(
+        Self,
+        Option<wasmer_runtime_core::codegen::DebugMetadata>,
+        HandlerData,
+    )> {
         let num_func_bodies = function_bodies.len();
         let mut local_relocs = Map::with_capacity(num_func_bodies);
         let mut external_relocs = Map::with_capacity(num_func_bodies);
 
         let mut trap_sink = TrapSink::new();
 
-        let compiled_functions: Result<Vec<(Vec<u8>, (RelocSink, LocalTrapSink))>, CompileError> =
-            function_bodies
-                .into_vec()
-                .par_iter()
-                .map_init(
-                    || Context::new(),
-                    |ctx, func| {
-                        let mut code_buf = Vec::new();
-                        ctx.func = func.to_owned();
-                        let mut reloc_sink = RelocSink::new();
-                        let mut local_trap_sink = LocalTrapSink::new();
-                        let mut stackmap_sink = NoopStackmapSink {};
-                        ctx.compile_and_emit(
-                            isa,
-                            &mut code_buf,
-                            &mut reloc_sink,
-                            &mut local_trap_sink,
-                            &mut stackmap_sink,
-                        )
-                        .map_err(|e| match e {
-                            CodegenError::Verifier(v) => CompileError::InternalError {
-                                msg: format!("Verifier error: {}", v),
-                            },
-                            _ => CompileError::InternalError { msg: e.to_string() },
-                        })?;
-                        ctx.clear();
-                        Ok((code_buf, (reloc_sink, local_trap_sink)))
-                    },
-                )
-                .collect();
+        let generate_debug_info = info.generate_debug_info;
+        let fb = function_bodies.iter().collect::<Vec<(_, _)>>();
+
+        /// Data about the the compiled machine code.
+        type CompileMetadata = (
+            LocalFuncIndex,
+            Option<(CompiledFunctionData, ValueLabelsRanges, Vec<Option<i32>>)>,
```

Comment:
Uh, it came from the stackslot type that `wasm-debug` used to depend on.  The offset it returns is a number.  I believe they're usually negative numbers actually because they're in terms of RBP and not RSI 🤔